### PR TITLE
refactor: removed filter flag switches. #1718.

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
### Reviewers
**Functional:** 
@tihuan 

---

### Changes
- Removed code switches related to filter flag.

### Notes
- Deleted deprecated collection detail `DatasetsGrid` component(frontend/src/components/Collections/components/Grid/components/DatasetsGrid/index.tsx) and moved related components (e.g. `DeleteDataset`, `MoreDropdown`, `RevisionStatusTag` etc.) to sibling of new collection detail `DatasetsGrid` (these moved files are marked as "File renamed without changes." in diff view).

### CODE REVIEW

File: frontend/src/common/queries/collections.ts [hunk 1]
```diff
@@ -201,14 +201,6 @@ export function useCollection({
     () => queryFn(id),
     {
       enabled: !!collections,
-      // TODO review use of refetch flag below on remove of filter flag (#1718). This is set to false as a fix for #2204
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/collections.ts:204]: A [defect](https://github.com/chanzuckerberg/single-cell-data-portal/issues/2204) was introduced when the filter flag functionality was added to collection form. `refetchOnWindowFocus: false` was added as a temporary fix until the filter flag function was removed.

File: frontend/src/common/hooks/useFeatureFlag.ts [hunk 2]
```diff
@@ -24,12 +24,6 @@ export function useFeatureFlag(
 
   /* Update state of enabled flag and redirect user if feature is not available to them. */
   useEffect(() => {
-    // Force enable of filter - start. Remove with #1718.
```
[the remark this answers is not in the document]
Comment (tihuan) [reply frontend/src/common/hooks/useFeatureFlag.ts:27]: Thanks for the context!!

File: frontend/tests/features/collection/collection.test.ts [hunk 3]
```diff
@@ -20,29 +19,6 @@ const TEST_COLLECTION = {
 };
 
 describe("Collection", () => {
-  describe("Collection Creation Modal", () => {
```
Comment (tihuan) [inline frontend/tests/features/collection/collection.test.ts:23]: Thanks for deleting these unit tests in the smoke test! IIRC the DOI validation is now done in BE?

Thank you!

**Review by tihuan — APPROVED** [review]
LGTM! Such a satisfying PR for consolidating pages/components and deleting code 🔥 🔪 ✨ !! Thanks so much for the amazing work 🙏 🌈

**Pull request merged**